### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by torobucci**
#### Create Interface
- Create a class `Nameable`.
   - Implement a method called `correct_name` that will raise a `NotImplementedError`.

#### Turn Person class to Nameable
- Make sure that your Person class inherits from Nameable
- Make sure that this class has a method `correct_name` implemented. It should simply return the name attribute.

#### Prepare base Decorator
- Make sure that it inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable.
- Implement the `correct_name` method that returns the result of the correct_name method of the `@nameable`.

#### Prepare CapitalizeDecorator and TrimmerDecorator
- CapitalizeDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
- TrimmerDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```
Comment (thecodechaser) [inline base_decorator.rb]: - [ ] Please make sure the `nameable` variable is readable and writable so decorate pattern works as expected.
   - You can consider using this `attr_accessor :nameable`.

File: capitalize_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require_relative './base_decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    super.capitalize
```
Comment (thecodechaser) [inline capitalize_decorator.rb]: - [ ] Please change the above code with this to work capitalize pattern to work as expected. `@nameable.correct_name.capitalize`

File: trimmer_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require_relative './base_decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    super[0..9]
```
Comment (thecodechaser) [inline trimmer_decorator.rb]: - [ ] Please change the above code with this to work the trimmer pattern to work as expected. `@nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name`.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @torobucci👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @torobucci,

✔️ Implementation of class decorator

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**